processing/src/main/java/io/druid/query/topn/AlphaNumericTopNMetricSpec.java

```diff
@@ -186,7 +186,8 @@ public AlphaNumericTopNMetricSpec(
       @JsonProperty("previousStop") String previousStop
   )
   {
-    super(previousStop);
+    // Set empty string as previousStop of super class otherwise we will get NPE
+    super((previousStop == null) ? "" : previousStop);
     this.previousStop = (previousStop == null) ? "" : previousStop;
```

Comment:
instead, can you put
this.previousStop = (previousStop == null) ? "" : previousStop;
in the constructor of LexicographicTopNMetricSpec

and here , replace line#191 by this.previousStop = super.getPreviousStop() ?